source/manual/howto-ssh-to-machines-in-aws.html.md

```diff
@@ -4,81 +4,133 @@ title: SSH into AWS machines
+Clone the `govuk-aws` repository and add a symlink to make `govukcli` executable globally:
 
-```
+```sh
 cd ~/govuk
 git clone https://github.com/alphagov/govuk-aws
+ln -s ~/govuk/govuk-aws/tools/govukcli /usr/local/bin/govukcli
 ```
 
-Add a symlink to ensure you use the current version:
+You will also need to have followed the [Get SSH access to integration](/manual/get-ssh-access.html)
+instructions.
+
+## Usage
+
+### Local dev machine
+
+To view all possible contexts, run:
 
+`local$ govukcli list-contexts`
+
+To set a persistent context:
+
+`local$ govukcli set-context integration`
+
+Now when you SSH into an instance, it will be to one running on the specified environment:
+
+```sh
+# SSH into a 'calculators_frontend' node in current context (Integration)
+local$ govukcli ssh calculators_frontend
 ```
-ln -s ~/govuk/govuk-aws/tools/govukcli /usr/local/bin/govukcli
+
+Alternatively, you can set a context for the current command only by passing a CLI parameter:
+
+```sh
+# SSH into a 'calculators_frontend' node in Staging, ignoring local context
+local$ govukcli ssh --context staging calculators_frontend
 ```
 
-## Connecting to instances
+Note that the SSH examples above will SSH you into a random machine of the right node class.
+You can find out which class you need by finding the corresponding [app page](https://docs.publishing.service.gov.uk/apps.html).
+
+`govukcli` finds a matching machine via the 'jumpbox', which you can also SSH into directly:
+
+`local$ govukcli ssh jumpbox`
+
+### Jumpbox
+
+The jumpbox is a special node that knows about all of the other nodes in its environment.
 
-The tool uses the idea of "contexts", where a "context" is a specific environment.
+List the IP addresses of every node in the environment:
 
-To view the contexts run:
+`jumpbox$ govuk_node_list`
 
-`govukcli list-contexts`
+This long list of IPs is not very useful on its own, but you can filter it by node class:
 
-Set a context to interact with that environment:
+`jumpbox$ govuk_node_list -c backend`
 
-`govukcli set-context integration`
+And if you can't remember the names of the node classes, there's a built-in helper:
 
-### AWS
+`jumpbox$ govuk_node_list --classes`
 
-In AWS you must specify a node class only. You must use `_` rather than `-` when specifying
-the class.
+Once you have found the IP of the machine you want to SSH into, you can manually SSH
+directly from the jumpbox machine:
 
-`govukcli ssh calculators_frontend`
+`jumpbox$ ssh ip-10-1-5-22.eu-west-1.compute.internal`
 
-This will take you to a random instance of that class.
+You can also do this from your local machine by appending the environment to the address:
 
-If you wish to go to a specific node, you can log into the jumpbox:
+`local$ ssh ip-10-1-5-22.eu-west-1.compute.internal.integration`
 
-`govukcli ssh jumpbox`
+Note that in Carrenza, there are no dynamic IPs, so you can SSH into a specific node by name:
+
+```sh
+# SSH into the `backend-1.backend.integration` node
```

Comment:
`backend-1.backend.integration` doesn't exist anymore, perhaps using `backend-1.backend.staging` instead for the example.

Reply:
Done 👍 have added an intro note just before it in case people are confused as to why we're suddenly giving examples for staging instead of integration.